retry.go

```diff
@@ -47,6 +49,42 @@ func X(x int, maxBackoff time.Duration, f func() bool) {
 	}
 }
 
+// XWithContext runs function f until f returns nil or the
+// number of retries exceeds x. Never more than x calls of f
+// are done. Calls to f have a sleep duration between them.
+// XWithError will return a wrapped error around f's errors
```

Comment:
copy pasta?

Reply:
I forgot to change the comment after changing the name. Had a hard time naming it.